pandas/tools/merge.py

```diff
@@ -894,13 +896,21 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x.values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return Series(new_data, index=self.new_axes[0], name=name).__finalize__(self, method='concat')
+                if self._is_sp_series:
```

Comment:
don't like all of these if-thens. The basic problem is the existance of ``SparseSeries/SparseDataFrame``, which don't really need to exist at all. The blocks actually hold the sparse data. But that is a bit bigger project.